nixos/modules/services/games/archisteamfarm.nix

```diff
@@ -193,7 +193,7 @@ in
             Group = "archisteamfarm";
             WorkingDirectory = cfg.dataDir;
             Type = "simple";
-            ExecStart = "${lib.getExe cfg.package} --no-restart --process-required --service --system-required --path ${cfg.dataDir}";
+            ExecStart = "${lib.getExe cfg.package} --no-config-migrate --no-restart --process-required --service --system-required --path ${cfg.dataDir}";
```

Comment:
This must be conditional on the bots being defined in the nixos module. You can run the service without it and then it should migrate the config files.

Also I am not sure how we want to handle config migration for declarative bots. We should at least mention it in the option for it.

Reply:
> This must be conditional on the bots being defined in the nixos module. You can run the service without it and then it should migrate the config files.

Honestly, I didn't even think that defining non-declarative bots was supported for this service. Either way, `ASF.json` is always defined declaratively, and `--no-config-migrate` also affects that. Even if the user used non-declarative bot configs, the issue of ASF migrating Nix-generated files is still an issue.

As far as I'm aware, ASF doesn't provide a means of enabling/disabling migration on a per-file basis, so this is an all-or-nothing option (unless someone wants to make a PR to ASF directly for that feature), so the only way to fix migration of Nix-generated files is to disable migration entirely.

> Also I am not sure how we want to handle config migration for declarative bots. We should at least mention it in the option for it.

I'm not sure *why* making it optional is warranted. "Not overwriting Nix-generated files" seems to be a de-facto requirement based on various comments I've seen in various packages and services, so I don't see why it should be optional here.

Also, "migration" really doesn't seem to affect much in terms of ASF's functionality. The main thing it seems to do is remove entries that assign a property to its default value (eg. if I define `"Enabled": false,`, migration will remove that entry since bots are disabled by default), which doesn't provide much benefit to the user. I'm not sure leaving it disabled by default will be a significant problem for any users.

Reply:
> Honestly, I didn't even think that defining non-declarative bots was supported for this service.

Why shouldn't it be? When creating them via the web ui, it just works and since some settings are a bit cryptic, I think we should continue to support this.

> Even if the user used non-declarative bot configs, the issue of ASF migrating Nix-generated files is still an issue.

Is it? I am running that setup for a long time and never encountered any issues with it.

> As far as I'm aware, ASF doesn't provide a means of enabling/disabling migration on a per-file basis, so this is an all-or-nothing option (unless someone wants to make a PR to ASF directly for that feature), so the only way to fix migration of Nix-generated files is to disable migration entirely.

Don't make it complicated. ASF.json is not readonly to susbtitute the secret, so this is only an issue for declarative bots.

> I'm not sure _why_ making it optional is warranted. "Not overwriting Nix-generated files" seems to be a de-facto requirement based on various comments I've seen in various packages and services, so I don't see why it should be optional here.

other services might have other reasons but eg nextcloud does allow it in a way more complicated way. Either way, as long as we support imperative bots, we should keep this.

> Also, "migration" really doesn't seem to affect much in terms of ASF's functionality. The main thing it seems to do is remove entries that assign a property to its default value (eg. if I define "Enabled": false,, migration will remove that entry since bots are disabled by default), which doesn't provide much benefit to the user.

Because there is currently nothing to migrate. I am to lazy to search but I think the migration works by reading in the old config, doing the migration in code somewhere and then writing it out. If doesn't change any values, you're config is up to date.